allennlp/models/biaffine_dependency_parser.py

```diff
@@ -490,9 +490,22 @@ def _mst_decode(self,
 
         # Shape (batch_size, sequence_length, sequence_length)
         normalized_arc_logits = F.log_softmax(attended_arcs, dim=2).transpose(1, 2)
+
+        # Although we need to include the root node so that the MST includes it,
+        # we do not want any word to be the parent of the root node.
+        # Here, we enforce this by setting the scores for all word -> ROOT edges
+        # edges to be very negative.
+        normalized_arc_logits[:, 0, :] = -1e8
+
         # Shape (batch_size, num_head_tags, sequence_length, sequence_length)
+        # This energy tensor expresses the following relation:
+        # energy[i,j] = "Score that j is the head of i". In this
```

Comment:
I think something's backwards in either this comment or the logic above.  Because if I substitute `j` for `ROOT` and `i` for `some_word`, I get "Score that `ROOT` is the head of `some_word`", which you set to very negative above, with `energy[:, 0, :] = -1e8`.  Right?

Reply:
I'm just saying there's a mismatch between your comment and your code.  Oh, wait, no, I was confusing myself because of the batch dimension.  When I wrote my comment, I was thinking it was `normalized_arc_logits[i, j, :]`, so `j` was `ROOT`, but it's actually `normalized_arc_logits[:, i, :]`, so `i` is `ROOT`.  Ok, all good.